staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/promise/promise_test.go

```diff
@@ -17,105 +17,107 @@ limitations under the License.
 package promise
 
 import (
-	"sync"
-	"sync/atomic"
+	"context"
 	"testing"
 	"time"
 
-	testclock "k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/testing/eventclock"
+	"k8s.io/apimachinery/pkg/util/wait"
 )
 
-func TestLockingWriteOnce(t *testing.T) {
-	now := time.Now()
-	clock, counter := testclock.NewFake(now, 0, nil)
-	var lock sync.Mutex
-	wr := NewWriteOnce(&lock, counter)
-	var gots int32
-	var got atomic.Value
-	counter.Add(1)
+func TestWriteOnceSet(t *testing.T) {
+	oldTime := time.Now()
+	cval := &oldTime
+	ctx, cancel := context.WithCancel(context.Background())
+	wr := NewWriteOnce(nil, ctx.Done(), cval)
+	gots := make(chan interface{})
 	go func() {
-		lock.Lock()
-		defer lock.Unlock()
-
-		got.Store(wr.Get())
-		atomic.AddInt32(&gots, 1)
-		counter.Add(-1)
+		gots <- wr.Get()
 	}()
-	clock.Run(nil)
-	time.Sleep(time.Second)
-	if atomic.LoadInt32(&gots) != 0 {
+	select {
```

Comment:
Let's maybe wrap this select also into a function (it's also used in few places).

Something like `expectNoGotValue`

Reply:
done